pkg/asset/agent/image/agentimage.go

```diff
@@ -105,29 +134,72 @@ func (a *AgentImage) updateIgnitionImg(ignition []byte) error {
 		return err
 	}
 
-	ignitionImgPath := filepath.Join(a.tmpPath, "images", "ignition.img")
-	fi, err := os.Stat(ignitionImgPath)
-	if err != nil {
-		return err
+	// Gathering the ignition image details
+	var (
+		ignImagePath                 string
+		ignStartOffset, ignMaxLength int64
+	)
+	ignInfoJSONPath := filepath.Join(a.tmpPath, "coreos", "igninfo.json")
+	if _, err := os.Stat(ignInfoJSONPath); err == nil {
+		// Fetching the ignition details from the igninfo file
+		err = a.fetchIgnitionInfo(ignInfoJSONPath)
+		if err != nil {
+			return err
+		}
+		ignImagePath = filepath.Join(a.tmpPath, ignInfo.File)
+		ignStartOffset = ignInfo.Offset
+		ignMaxLength = ignInfo.Length
+	} else {
+		// Defaulting the ignition image path if no info file is found
+		ignImagePath = filepath.Join(a.tmpPath, "images", "ignition.img")
+		ignStartOffset = 0
+		ignMaxLength = 0
```

Comment:
It could be safe to set `ignMaxLength` here equal to `fileInfo.Size()`. This will allow to simplify a little bit the code below and have a single capacity check

Reply:
@phani2898  better to remove the padding logic.

Reply:
@andfasano , The fileInfo which you are indicating here is ignition.img ? So are you suggesting to keep the maxlength always aligned to fileSize ?